app/api/domains/api.py

```diff
@@ -107,6 +109,40 @@ def format_map_basic(m: Beatmap) -> dict[str, object]:
         "diff": m.diff,
     }
 
+@router.get("/search")
+async def api_search(
+    search: Optional[str] = Query(None, alias='src'),
+    db_conn: databases.core.Connection = Depends(acquire_db_conn),
+):
+    """Search for users on the server by name."""
+    
+    # we're using text() here because we are going to
+    # assign placeholders (bindparams()) to these queries, 
+    # and this will let us bind those safely and efficiently.
+    results_query = text("SELECT COUNT(id) FROM users WHERE name AND priv >= 3 LIKE :search_clause") # we will only need to lookup users who have Privileges.VERIFIED
+    rows_query = text("SELECT id, name FROM users WHERE name LIKE :search_clause AND priv >= 3 ORDER BY id ASC") # to save on host resources and prevent useless requests
+
+    # using parameterised queries helps prevent against sql injections
+    # and will also allow the database engine to cache our queries,
+    # making these lookups faster in the future
+    search_clause = f"%{search}%" if search else None
+    results_query = results_query.bindparams(search_clause=search_clause)
```

Comment:
why not let the database handle parameter binding?

Reply:
using bindparams() lets me bind the values safely, for when i am constructing queries dynamically (such as in this route). this is the safest way to avoid SQL injection, and improves performance on search because the engine will cache our queries.

in contrast, concatenating values directly into the query string can open us up for SQL injection, and makes our queries slower and uncacheable. if you have a different approach that avoids the latter, i would love to hear it

Reply:
Check the code on the server...

Reply:
you can just do
```py
await db_conn.fetch_val("SELECT name FROM users WHERE id = :id", {"id":1})
```
for example

Reply:
i don't agree with this approach, because in most cases with this route, we're executing multiple queries handled by a frontend where `fetch_val()` is slower overall than `bindparams()`

if we're going to be dynamically constructing a query, then `bindparams()` is the way i want to do this.
if we are just going to be executing a singular query with fixed values, then i would agree with `fetch_val()`

Reply:
> i don't agree with this approach, because in most cases with this route, we're executing multiple queries handled by a frontend where `fetch_val()` is slower overall than `bindparams()`
> 
> if we're going to be dynamically constructing a query, then `bindparams()` is the way i want to do this.
> if we are just going to be executing a singular query with fixed values, then i would agree with `fetch_val()`
> 
> both could be equally good towards this, i would like some more input before i make any changes though

fetch_val was an example. It works on any fetch or execute method.

Reply:
Then rewrite all of bpy so everything is uniform again

Reply:
hey, the databases library were using is already doing this stuff behind the scenes for you - no need to use it directly here

Reply:
sure, we'll use it